File: generate-icons.js
// Simple script to generate icon files using Node.js
// Run with: node generate-icons.js

const fs = require('fs');
const path = require('path');

// Create a simple SVG icon that can be converted to PNG
// For now, we'll create a simple data URL approach
// Note: This requires a browser or image processing library
// For a quick solution, users can use the generate-icons.html file

console.log('Icon generation script');
console.log('Please use generate-icons.html in a browser to generate the icon files.');
console.log('Or install a package like "canvas" and run this script with proper image generation.');

// Alternative: Create a simple script that uses canvas if available
try {
  const { createCanvas } = require('canvas');
  const iconsDir = path.join(__dirname, 'icons');
  
  if (!fs.existsSync(iconsDir)) {
    fs.mkdirSync(iconsDir, { recursive: true });
  }
  
  function drawIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Background
    ctx.fillStyle = '#3B82F6'; // Blue
    ctx.fillRect(0, 0, size, size);
    
    // Draw a simple ping/radar icon
    const centerX = size / 2;
    const centerY = size / 2;
    const radius = size * 0.35;
    
    // Outer circle
    ctx.strokeStyle = '#FFFFFF';
    ctx.lineWidth = size * 0.08;
    ctx.beginPath();
    ctx.arc(centerX, centerY, radius, 0, Math.PI * 2);
    ctx.stroke();
    
    // Inner circle
    ctx.beginPath();
    ctx.arc(centerX, centerY, radius * 0.6, 0, Math.PI * 2);
    ctx.stroke();
    
    // Center dot
    ctx.fillStyle = '#FFFFFF';
    ctx.beginPath();
    ctx.arc(centerX, centerY, size * 0.1, 0, Math.PI * 2);
    ctx.fill();
    
    // Ping lines (radar sweep)
    ctx.strokeStyle = '#FFFFFF';
    ctx.lineWidth = size * 0.05;
    for (let i = 0; i < 4; i++) {
      const angle = (Math.PI * 2 * i) / 4;
      ctx.beginPath();
      ctx.moveTo(centerX, centerY);
      ctx.lineTo(
        centerX + Math.cos(angle) * radius,
        centerY + Math.sin(angle) * radius
      );
      ctx.stroke();
    }
    
    return canvas;
  }
  
  // Generate icons
  [16, 48, 128].forEach(size => {
    const canvas = drawIcon(size);
    const buffer = canvas.toBuffer('image/png');
    const filePath = path.join(iconsDir, `icon${size}.png`);
    fs.writeFileSync(filePath, buffer);
    console.log(`Generated icon${size}.png`);
  });
  
  console.log('All icons generated successfully!');
} catch (error) {
  console.log('Canvas package not found. Please either:');
  console.log('1. Use generate-icons.html in a browser to generate icons');
  console.log('2. Install canvas: npm install canvas');
  console.log('3. Or create your own icon files manually');
}

File: generate-icons.py
#!/usr/bin/env python3
"""
Generate icon files for the Chrome extension.
Requires Pillow: pip install Pillow
"""

try:
    from PIL import Image, ImageDraw
    import os
    
    def create_icon(size):
        # Create a new image with blue background
        img = Image.new('RGB', (size, size), color='#3B82F6')
        draw = ImageDraw.Draw(img)
        
        center_x = size // 2
        center_y = size // 2
        radius = int(size * 0.35)
        
        # Draw outer circle
        outer_bbox = [
            center_x - radius,
            center_y - radius,
            center_x + radius,
            center_y + radius
        ]
        draw.ellipse(outer_bbox, outline='white', width=max(1, int(size * 0.08)))
        
        # Draw inner circle
        inner_radius = int(radius * 0.6)
        inner_bbox = [
            center_x - inner_radius,
            center_y - inner_radius,
            center_x + inner_radius,
            center_y + inner_radius
        ]
        draw.ellipse(inner_bbox, outline='white', width=max(1, int(size * 0.08)))
        
        # Draw center dot
        dot_radius = max(1, int(size * 0.1))
        dot_bbox = [
            center_x - dot_radius,
            center_y - dot_radius,
            center_x + dot_radius,
            center_y + dot_radius
        ]
        draw.ellipse(dot_bbox, fill='white')
        
        # Draw ping lines (radar sweep)
        import math
        line_width = max(1, int(size * 0.05))
        for i in range(4):
            angle = (math.pi * 2 * i) / 4
            end_x = center_x + int(math.cos(angle) * radius)
            end_y = center_y + int(math.sin(angle) * radius)
            draw.line([center_x, center_y, end_x, end_y], fill='white', width=line_width)
        
        return img
    
    # Create icons directory if it doesn't exist
    icons_dir = 'icons'
    os.makedirs(icons_dir, exist_ok=True)
    
    # Generate icons
    for size in [16, 48, 128]:
        icon = create_icon(size)
        icon_path = os.path.join(icons_dir, f'icon{size}.png')
        icon.save(icon_path, 'PNG')
        print(f'Generated {icon_path}')
    
    print('All icons generated successfully!')
    
except ImportError:
    print('Pillow is not installed. Please install it with: pip install Pillow')
    print('Or use generate-icons.html in a browser to generate the icons.')
except Exception as e:
    print(f'Error generating icons: {e}')
    print('You can also use generate-icons.html in a browser to generate the icons.')

